toolchain/sem_ir/typed_insts.h

```diff
@@ -910,6 +910,27 @@ struct SpecificConstant {
   SpecificId specific_id;
 };
 
+// A specific instance of a generic function. This represents the callee in a
+// call instruction that is calling a generic function, where the specific
+// arguments of the function have been deduced.
+//
+// TODO: This value corresponds to the `(FunctionType as Call(...)).Op` function
+// in the overloaded calls design. Eventually we should represent it more
+// directly as a member of the `Call` interface.
+struct SpecificFunction {
+  static constexpr auto Kind = InstKind::SpecificFunction.Define<Parse::NodeId>(
```

Comment:
`Define<Parse::NodeID>` usually seems to be accompanied by a TODO to make the node type more specific. Should that be done here?

Reply:
No -- the `SpecificFunction` has the location of the callee in the call expression, which is an arbitrary expression, so we can't associate it with a more specific parse node.